include/nbl/builtin/hlsl/memory_accessor.hlsl

```diff
@@ -257,6 +257,36 @@ struct MemoryAdaptor<BaseAccessor, 0>
     }
 };
 
+// ---------------------------------------------- Offset Accessor ----------------------------------------------------
+
+template<class BaseAccessor, class AccessorType, uint32_t Offset>
+struct OffsetAccessor
+{
+    BaseAccessor accessor;
+
+    void set(uint32_t idx, NBL_REF_ARG(AccessorType) x) {accessor.set(idx + Offset, x);}
+
+    AccessorType get(uint32_t idx) {return accessor.get(idx + Offset);}
```

Comment:
you can work out the types using `using value_type = decltype(declval<BaseAccessor>().get(0xdeadbeefu))`

although due to the presence of overloads, I'd make the set and get possibly templated?

Reply:
why do you make the `get` signature different from all the above!?